host/class/uac/usb_host_uac/uac_host.c

```diff
@@ -199,6 +202,34 @@ static esp_err_t _uac_host_device_delete(uac_device_t *uac_device);
 static esp_err_t uac_cs_request_set(uac_device_t *uac_device, const uac_cs_request_t *req);
 static esp_err_t uac_cs_request_set_ep_frequency(uac_iface_t *iface, uint8_t ep_addr, uint32_t freq);
 
+// --------------------------- Utility Functions --------------------------------
+/**
+ * @brief Convert volume U16 type to UAC volume db, for details, please refer to UAC 1.0 spec P.77
+ * 0x7FFF: 127.9961db
+ * 0x0100: 1.0000db
+ * 0x0000: 0.0000db
+ * 0xFF00: -1.0000db (there is an typo in UAC 1.0 spec, it should be 0xFF00)
+ * 0xFE00: -2.0000db
+ * 0x8001: -127.9961db
+ * @param[in] volume      Volume in db
+ * @return uint16_t
+ */
+static float _volume_db_u16_2_f(uint16_t volume_db)
+{
+    return (volume_db & 0x8000 ? -128.0 : 0) + ((uint16_t)(volume_db & 0x7fff)) * 1.0 / 256;
+}
+
+static uint16_t _volume_db_f_2_u16(float volume_db_f)
+{
+    uint16_t volume_db = 0;
```

Comment:
When we have several arithmetic operation in sequence, sometimes it is better to split them logically and apply masks (if necessary) with specific flags. Also, it helps to verify the edge or overflow cases. 

Something like: 

```suggestion
    bool negative = false;
    if(volume_db_f < 0) {
        volume_db_f += 128.0; // could here be an overflow ? 
        negative = true;
    }
    
    volume_db = (uint16_t)round(volume_db_f * 256); // could here be an overflow? 
    
    return (negative)? volume_db | 0x8000 : volume_db;
    
```

Reply:
@roma-jam Thanks for your advice. I have changed to using int16_t to avoid manual manipulation of negative numbers. c2a3782b0f011edf2fb78814175ede1b035d27f8

Add I also add the check to avoid overflow fc3ca5d53d49b100f6f2ffe3402f9c4968b1da4a

Reply:
The value 127.9961 is from the UAC 1.0 spec

![image](https://github.com/user-attachments/assets/a90e0a95-f788-4ce6-b5ba-1fc71857d65e)